src/inspectCommits.ts

```diff
@@ -77,22 +78,27 @@ export async function inspectCommits(pkg: Package, lerna = false): Promise<Commi
 
   const releasableCommits: Commit[] = [];
   const unreleasableCommits: Commit[] = [];
+  let isMajorBump = false;
   for (const commit of commits) {
     const headerIndicatesMajorChange = !!commit.header && commit.header.includes('!');
     const bodyIndicatesMajorChange = !!commit.body && commit.body.includes('BREAKING');
     const typeIsSkippable = skippableCommitTypes.includes(commit.type);
-    const isReleasable = !typeIsSkippable || bodyIndicatesMajorChange || headerIndicatesMajorChange;
+    const isMajorChange = bodyIndicatesMajorChange || headerIndicatesMajorChange;
+    const isReleasable = !typeIsSkippable || isMajorChange;
     if (isReleasable) {
       releasableCommits.push(commit);
     } else {
       unreleasableCommits.push(commit);
     }
+
+    if (isMajorChange && !isMajorBump) isMajorBump = true;
```

Comment:
Won't `!isMajorBump` always be `true` here? If so what's the point of including it?

Reply:
@RodEsp `isMajorBump` is `false` until it's been set to `true`

The idea is that once that it's been set to `true`, we don't want it to get set back to `false`

Reply:
@mdonnalley I guess I'm confused because there doesn't seem to be anything between line 81, where `let isMajorBump = false;` happens and this line that would change `isMajorBump`. So in this check `!isMajorBump` always evaluates to `true`.

Am I missing something?

Reply:
@RodEsp Yes, I think you're missing that this code is iterating over an array of commits.

On line 81, I set `isMajorBump` to `false` to before iterating over the commits

On line 86, I figure out if the current commit requires a major bump (`isMajorChange`)

On line 94, `isMajorBump` is set to true if it's currently false and `isMajorChange` is true

The purpose `!isMajorBump` is to ensure that no subsequent commits change it back to `false` after it's been set to `true`

Reply:
So iterating over the commits might make `isMajorChange` switch to a different value?
If so can you add a comment about how that works and why `!isMajorChange` is used here please?

Reply:
@mdonnalley ☝🏼 fyi